### ISSUE TITLE
Visual tearing on background grid

### ISSUE THREAD

**Issue by mjsutcliffe99** [zxcalc/zxlive#341]
I'm not sure if this has already been reported, but I sometimes find the background grid gets all janky with some strange tearing going on, as pictured, when scrolling beyond frame. It's very superficial and not a big deal (zooming in and out seems to be an easy quick fix).

![tearing](https://github.com/user-attachments/assets/58942d8f-0009-4d9a-9baf-04aceea83737)

**Comment by RazinShaikh** [comment]
This might be due to caching the background but I have not verified this.
https://github.com/zxcalc/zxlive/blob/master/zxlive%2Fgraphview.py#L88

**Comment by jvdwetering** [comment]
I've noticed this mostly happens when you do a combined horizontal and vertical scroll using for instance your laptop touchpad. If you scroll regularly this seems to happen not at all or at least way less.

**Comment by RazinShaikh** [comment]
I tried fixing it but none of what I tried worked. I am tempted to say that it might be a QT issue.

**Comment by jvdwetering** [comment]
It's not an issue in Tikzit though, which also uses Qt. Maybe something to ask on Stackexchange?

**Comment by dlyongemallo** [comment]
This seems to be due to the combination of background caching and the bounding rect viewport update mode. This can be fixed by just dropping `setCacheMode(CacheBackground)` or switching the viewport update mode to `FullViewportUpdate` or `SmartViewportUpdate`. Depends on what tradeoff you want to make.

If it's only or mostly a problem for diagonal scrolling, the `scrollContentsBy` could be overridden to reset the cache whenever `dx` and `dy` are both non-zero.

**Comment by jvdwetering** [comment]
Is there any performance hit to dropping caching?

**Comment by dlyongemallo** [comment]
Yes, it would mean `drawBackground` is run on every paint. This can add up during rapid scrolling or when zoomed out on a large graph.

I think resetting the cache only on diagonal scrolling, and maybe expanding the `drawBackground` slightly beyond the clip region, would be enough to mostly fix the problem without completely dropping caching.

**Comment by dlyongemallo** [comment]
> I tried fixing it but none of what I tried worked.

@RazinShaikh Can you give #546 a try and let me know if that fixes it for you, or if not, if there are any noticeable differences?
